tests/MongoDB.Bson.Tests/HexTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using FluentAssertions;
+using MongoDB.Shared;
+using Xunit;
+
+namespace MongoDB.Bson.Tests
+{
+    public class HexTests
+    {
+        [Theory]
+        [InlineData(0)]
+        [InlineData('/')]
+        [InlineData(':')]
+        [InlineData('@')]
+        [InlineData('G')]
+        [InlineData('`')]
+        [InlineData('g')]
+        [InlineData(127)]
+        [InlineData(255)]
+        [InlineData(0xffff)]
```

Comment:
I guess 0xffff is a valid hex digit. Maybe we can think about another name for this test?

Reply:
`0xffff` is NOT a valid hex digit.

`0xffff` is the same as `char.MaxValue`.

Reply:
Sorry @rstam, I'm not a big expert in this part.
I just thought that we can convert any digit to `hex` form. 
Like:
int a = 0x1A2F;           // 6703
int  b = 0xFFFF;          // 65535
int c = 0x1234;           // 4660

Reply:
maybe I don't see something

Reply:
Your examples aren't hex strings as used by us, they are hex integer literals in the C# language.

For our purposes we are defining:

- a hex string is a string of zero or more hex digits
- a hex digit is a char that is one of: '0' through '9', 'a' through 'f' or 'A' through 'F'